internal/linker/link_node_modules.bzl

```diff
@@ -36,16 +37,21 @@ def add_arg(args, arg):
 
 def _link_mapping(label, mappings, k, v):
     if k in mappings and mappings[k] != v:
+        # Allow all other mappings to win over legacy "_tslibrary"
```

Comment:
why?

Reply:
Oh. Such a long story. I can get rid of it as a breaking change by adding package_name to ts_library for linking and reserving module_name for the AMD name in the UMD/AMD output.

I left a paragraph comment in the code about tthis.
```
    # Special case for ts_library module_name for legacy behavior and for AMD name work-around
    # Tag the mapping as "_tslibrary" so it can be overridden by any other mapping if found.
    #
    # In short, ts_library module_name attribute results in both setting the AMD name (which
    # desired and necessary in devmode which outputs UMD) and in making a linkable mapping. Because
    # of this, you can get in the situation where a ts_library module_name and a downstream pkg_name
    # package_name conflict and result in duplicate mappings. This work-around will make this
    # situation work however it is not a recommended pattern since a ts_library can be a dep of a
    # pkg_npm but not vice-verse at the moment since ts_library cannot handle directory artifacts as
    # deps.
```